test/core/client_channel/lb_policy/lb_policy_test_lib.h

```diff
@@ -451,7 +465,31 @@ class LoadBalancingPolicyTest : public ::testing::Test {
     auto uri = grpc_sockaddr_to_uri(&subchannel->address());
     ASSERT_TRUE(uri.ok()) << uri.status() << " at " << location.file() << ":"
                           << location.line();
-    EXPECT_EQ(*uri, address_uri) << location.file() << ":" << location.line();
+    out_picked_uri = *uri;
+  }
+
+  // Requests a pick on picker and expects a Complete result whose
+  // subchannel has the specified address.
+  void ExpectPickComplete(LoadBalancingPolicy::SubchannelPicker* picker,
+                          absl::string_view address_uri,
+                          SourceLocation location = SourceLocation()) {
+    absl::optional<std::string> picked_uri;
+    GetPickedAddress(picker, picked_uri, location);
+    EXPECT_EQ(*picked_uri, address_uri)
+        << location.file() << ":" << location.line();
+  }
+
+  // Requests a pick on picker and expects a Fail result.
+  void ExpectPickFail(LoadBalancingPolicy::SubchannelPicker* picker,
```

Comment:
This method should check the status with which the LB policy fails the pick, since that status will wind up being used as the failure status of the data plane RPC, and we want to make sure that such statuses are properly helpful for our users.

I suggest adding a required parameter of type `std::function<void(absl::Status)>` that we will pass the failure status to.  The caller can pass in a lambda that checks the status code and value.

Reply:
I inlined this function as there had only been a single call site.